Update of ipython to 6.3.0 breaks unit tests

CI unit test build suddenly failing after recent update to ipython 6.3:
```
______________________________ test_Model_pretty _______________________________
    def test_Model_pretty():
>       class Foo1(Model):
bokeh/tests/test_model.py:8: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
meta_cls = <class 'bokeh.model.MetaModel'>, class_name = 'Foo1'
bases = (<class 'bokeh.model.Model'>,)
class_dict = {'__container_props__': set(), '__module__': 'bokeh.tests.test_model', '__properties__': set(), '__properties_with_refs__': set(), ...}
    def __new__(meta_cls, class_name, bases, class_dict):
        '''
    
            Raises:
                Warning
    
            '''
    
        # use an explicitly provided view model name if there is one
        if "__view_model__" not in class_dict:
            class_dict["__view_model__"] = class_name
    
        # call the parent metaclass to create the new model type
        newcls = super(MetaModel, meta_cls).__new__(meta_cls, class_name, bases, class_dict)
    
        # update the mapping of view model names to classes, checking for any duplicates
        # and handling any subtype relationships or custom implementations
        entry = class_dict.get("__subtype__", class_dict["__view_model__"])
        if entry in MetaModel.model_class_reverse_map and not hasattr(newcls, "__implementation__"):
            raise Warning("Duplicate __view_model__ or __subtype__ declaration of '%s' for " \
                          "class %s.  Previous definition: %s" % \
                          (entry, class_name,
>                          MetaModel.model_class_reverse_map[entry]))
E           Warning: Duplicate __view_model__ or __subtype__ declaration of 'Foo1' for class Foo1.  Previous definition: <class 'bokeh.tests.test_model.test_Model_pretty.<locals>.Foo1'>

```
confirmed locally that this package difference triggers the failure.

One option is to pin the version but would like to identify the root cause as this connection is surprising and unexpected.

Well this just gets more annoying, locally the same test is actually failing for a different reason:
```

    def test_Model_pretty():
        class Foo1(Model):
            pass
>       assert Foo1(id='1').pretty() == """\
    bokeh.tests.test_model.Foo1(
        id='1',
        js_event_callbacks={},
        js_property_callbacks={},
        name=None,
        subscribed_events=[],
        tags=[])"""
E       assert "Foo1(id='1', ...)" == 'bokeh.tests.test_m...s=[],\n    tags=[])'
E         - Foo1(id='1', ...)
E         + bokeh.tests.test_model.Foo1(
E         +     id='1',
E         +     js_event_callbacks={},
E         +     js_property_callbacks={},
E         +     name=None,
E         +     subscribed_events=[],
E         +     tags=[])

tests/test_model.py:10: AssertionError
```

--- a/bokeh/core/has_props.py
+++ b/bokeh/core/has_props.py
@@ -646,24 +646,18 @@ class HasProps(with_metaclass(MetaHasProps, object)):
 
         '''
         IPython = import_optional('IPython')
+        cls = self.__class__
 
         if IPython:
             from IPython.lib.pretty import RepresentationPrinter
 
-        class _BokehPrettyPrinter(RepresentationPrinter):
-            def __init__(self, output, verbose=False, max_width=79, newline='\n'):
-                super(_BokehPrettyPrinter, self).__init__(output, verbose, max_width, newline)
-                self.type_pprinters[HasProps] = lambda obj, p, cycle: obj._repr_pretty(p, cycle)
-
-        if not IPython:
-            cls = self.__class__
-            raise RuntimeError("%s.%s.pretty() requires IPython" % (cls.__module__, cls.__name__))
-        else:
             stream = StringIO()
-            printer = _BokehPrettyPrinter(stream, verbose, max_width, newline)
+            printer = RepresentationPrinter(stream, verbose, max_width, newline)
             printer.pretty(self)
             printer.flush()
             return stream.getvalue()
+        else:
+            raise RuntimeError("%s.%s.pretty() requires IPython" % (cls.__module__, cls.__name__))
 
     def pprint(self, verbose=False, max_width=79, newline='\n'):
         ''' Print a "pretty" string representation of the object to stdout.
@@ -723,7 +717,7 @@ class HasProps(with_metaclass(MetaHasProps, object)):
         '''
         return self.__class__(**self._property_values)
 
-    def _repr_pretty(self, p, cycle):
+    def _repr_pretty_(self, p, cycle):
         '''
 
         '''

--- a/bokeh/model.py
+++ b/bokeh/model.py
@@ -649,7 +649,7 @@ class Model(with_metaclass(MetaModel, HasProps, PropertyCallbackManager, EventCa
 
         return html
 
-    def _repr_pretty(self, p, cycle):
+    def _repr_pretty_(self, p, cycle):
         '''
 
         '''

